packages/fxa-auth-server/lib/payments/iap/apple-app-store/subscriptions.ts

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { IapExtraStripeInfo } from 'fxa-shared/dto/auth/payments/iap-subscription';
+import { MozillaSubscriptionTypes } from 'fxa-shared/subscriptions/types';
+import Container from 'typedi';
+
+import { internalValidationError } from '../../../../lib/error';
+import { AppConfig } from '../../../types';
+import { StripeHelper } from '../../stripe';
+import { SubscriptionsService } from '../types';
+import { AppleIAP } from './apple-iap';
+import { AppStoreSubscriptionPurchase } from './subscription-purchase';
+
+export class AppStoreSubscriptions
+  implements
+    SubscriptionsService<AppStoreSubscriptionPurchase & IapExtraStripeInfo>
+{
+  private appleIap?: AppleIAP;
+  private stripeHelper?: StripeHelper;
+
+  constructor() {
+    const config = Container.get(AppConfig);
+    if (!config.subscriptions.enabled) {
+      throw internalValidationError(
+        'AppStoreSubscriptions',
+        {},
+        new Error(
+          'Trying to new up AppStoreSubscriptions while subscriptions are disabled.  Check your dependency graph.'
+        )
+      );
+    }
+
+    if (Container.has(AppleIAP)) {
+      this.appleIap = Container.get(AppleIAP);
+    }
+
+    if (Container.has(StripeHelper)) {
+      this.stripeHelper = Container.get(StripeHelper);
+    }
```

Comment:
Is there a reason we'd want to allow subscriptions to be enabled, and AppleIAP to be enabled, but then ignore not having a StripeHelper? Why do we want to allow this object to be created if it can't do its job?

Reply:
Not a good reason, but this is how the existing Google Play `PlaySubscriptions` class' `constructor` is set up as well.

While we don't throw an error, we would not make any calls in `getSubscriptions`, since there is an early return with `[]` as the list of Apple IAP subscriptions if Stripe Helper were not present.

The only scenario I can think of would be for testing. Some FxA routes call into this class (or will), and some of their tests may enable subscriptions but not set a `config.subscriptions.stripeApiKey` as is required to set up the Stripe Helper. If they aren't testing subscriptions but only a routes class that happens to call into the subscriptions code on one or two routes, I wouldn't want their test to suddenly start throwing.

Are you suggesting I should change this check and the corresponding check in `PlaySubscriptions` to throw an error if there's no Stripe Helper instead?